observability/opentelemetry/src/main/java/org/wildfly/extension/opentelemetry/OpenTelemetrySubsystemDefinition.java

```diff
@@ -0,0 +1,168 @@
+            .setAllowExpression(true)
+            .setAttributeGroup(GROUP_EXPORTER)
+            .setRestartAllServices()
+            .setDefaultValue(new ModelNode("http://localhost:14250"))
+            .build();
+
+
+    public static final SimpleAttributeDefinition SPAN_PROCESSOR_TYPE = SimpleAttributeDefinitionBuilder
+            .create(OpenTelemetryConfigurationConstants.SPAN_PROCESSOR_TYPE, ModelType.STRING, true)
+            .setAllowExpression(true)
+            .setXmlName(OpenTelemetryConfigurationConstants.TYPE)
+            .setAttributeGroup(GROUP_SPAN_PROCESSOR)
+            .setRestartAllServices()
+            .setDefaultValue(new ModelNode("batch"))
+            .setAllowedValues(ALLOWED_SPAN_PROCESSORS)
+            .setValidator(new StringAllowedValuesValidator(ALLOWED_SPAN_PROCESSORS))
+            .build();
+
+    public static final SimpleAttributeDefinition BATCH_DELAY = SimpleAttributeDefinitionBuilder
+            .create(OpenTelemetryConfigurationConstants.BATCH_DELAY, ModelType.LONG, true)
+            .setAllowExpression(true)
+            .setAttributeGroup(GROUP_SPAN_PROCESSOR)
+            .setRestartAllServices()
+            .setDefaultValue(new ModelNode(5000))
+            .build();
+
+    public static final SimpleAttributeDefinition MAX_QUEUE_SIZE = SimpleAttributeDefinitionBuilder
+            .create(OpenTelemetryConfigurationConstants.MAX_QUEUE_SIZE, ModelType.LONG, true)
+            .setAllowExpression(true)
+            .setAttributeGroup(GROUP_SPAN_PROCESSOR)
+            .setRestartAllServices()
+            .setDefaultValue(new ModelNode(2048))
+            .build();
+
+    public static final SimpleAttributeDefinition MAX_EXPORT_BATCH_SIZE = SimpleAttributeDefinitionBuilder
+            .create(OpenTelemetryConfigurationConstants.MAX_EXPORT_BATCH_SIZE, ModelType.LONG, true)
+            .setAllowExpression(true)
+            .setAttributeGroup(GROUP_SPAN_PROCESSOR)
+            .setRestartAllServices()
+            .setDefaultValue(new ModelNode(512))
+            .build();
+
+    public static final SimpleAttributeDefinition EXPORT_TIMEOUT = SimpleAttributeDefinitionBuilder
+            .create(OpenTelemetryConfigurationConstants.EXPORT_TIMEOUT, ModelType.LONG, true)
+            .setAllowExpression(true)
+            .setAttributeGroup(GROUP_SPAN_PROCESSOR)
+            .setRestartAllServices()
+            .setDefaultValue(new ModelNode(30000))
+            .build();
+
+    public static final SimpleAttributeDefinition SAMPLER = SimpleAttributeDefinitionBuilder
+            .create(OpenTelemetryConfigurationConstants.SAMPLER_TYPE, ModelType.STRING, true)
+            .setAllowExpression(true)
+            .setXmlName(OpenTelemetryConfigurationConstants.TYPE)
+            .setAttributeGroup(GROUP_SAMPLER)
+            .setRestartAllServices()
+            .build();
+
+    public static final SimpleAttributeDefinition SAMPLER_ARG = SimpleAttributeDefinitionBuilder
```

Comment:
Isn't this too general? Can we rather call this ratio, set limits for the value, and set a default value?

Reply:
Perhaps, and I tend to agree with you, but this is what the otel spec calls it, so I left it as is in case it's ever used in the future for something else as well. I guess in that case we can just add another param on our side to cover it.

Reply:
Thanks for the explanation. What about a dynamic validator for the value then? In case the sampler is `ratio` just 0.0 - 1.0 will be allowed?

Reply:
Changed.